hydrogym/core.py

```diff
@@ -272,12 +276,14 @@ def __init__(self, flow: PDEBase, dt: float = None):
     if dt is None:
       dt = flow.DEFAULT_DT
     self.dt = dt
+    self.t = 0.0
```

Comment:
Currently the time is owned by the flow, not the solver.  I guess you could do it the other way around, but since the "state" (velocity, pressure, etc) is owned by the flow class I figured it made more sense to have time there as well.  Then it is advanced with `flow.advance_time(dt)` [here](https://github.com/dynamicslab/hydrogym/blob/8698bf0915417a5ab68942f0e992df8dd4e16035/hydrogym/core.py#L176).  On the other hand, the solver doesn't directly depend on time, only the timestep, so my feeling is that it doesn't naturally belong there.

Actually I think the only place it's needed is for the "callbacks" or "controller", but now that time is a property of the flow it doesn't need to be there at all (or the callback could be passed `flow.t` instead of calculating it within the loop).

Reply:
- I agree, a solver object can be viewed as a mapping taking as input the current time $t$ (for non-autonomous systems), the current state $x_t$ and a time duration $T$ to return the solution $x_{t+T}$ $T$ times later. (this mapping is called a flow in maths). But the mapping in itself is independent of the time! Hence, no need to initialise it with `self.t = 0.0`.

- Practically, I don't see any use of `self.t` in this class, so it could be safe to remove it?